src/H5.c

```diff
@@ -101,32 +100,24 @@ static H5_atclose_node_t *H5_atclose_head = NULL;
 /* Declare a free list to manage the H5_atclose_node_t struct */
 H5FL_DEFINE_STATIC(H5_atclose_node_t);
 
-/*--------------------------------------------------------------------------
-NAME
-    H5__init_package -- Initialize interface-specific information
-USAGE
-    herr_t H5__init_package()
-RETURNS
-    Non-negative on success/Negative on failure
-DESCRIPTION
-    Initializes any interface-specific data or routines.
---------------------------------------------------------------------------*/
-herr_t
-H5__init_package(void)
+static herr_t
+H5_default_vfd_init(void)
 {
-    herr_t ret_value = SUCCEED; /* Return value */
-
-    FUNC_ENTER_NOAPI_NOINIT
+    hid_t  id;
+    herr_t ret_value = SUCCEED;
 
-    /* Run the library initialization routine, if it hasn't already ran */
-    if (!H5_INIT_GLOBAL && !H5_TERM_GLOBAL) {
-        if (H5_init_library() < 0)
-            HGOTO_ERROR(H5E_LIB, H5E_CANTINIT, FAIL, "unable to initialize library")
-    } /* end if */
+    FUNC_ENTER_NOAPI(FAIL)
+    /* Load the hid_t for the default VFD for the side effect
+     * it has of initializing the default VFD.
+     */
+    id = H5FD_sec2_init();
 
+    if (id == H5I_INVALID_HID) {
+        HGOTO_ERROR(H5E_FUNC, H5E_CANTINIT, FAIL, "unable to load default VFD ID")
+    }
```

Comment:
The library idiom is to combine the assignment and the check on the same line. It's a longer code line, but it ensures that values don't slip by unchecked.

Reply:
No intermediate variable is necessary here.